extensions/amp-base-carousel/0.1/test/test-carousel.js

```diff
@@ -198,7 +204,9 @@ describes.realWin('carousel implementation', {}, (env) => {
         slideCount: 3,
         loop: false,
       });
-      carousel.goToSlide(1, {smoothScroll: false});
+
+      // Fake the scroll
```

Comment:
Why is this change necessary? If possible we should avoid overwriting internal state in tests.

Reply:
This change is necessary since we changed `isAtStart` & `isAtEnd` to also use current index in it's calculation and that only get updates once the scroll event occurs (via listener), which is hard to test. 

I'll change the tests to also ensure that the scroll position is also correct (since that's what `isAtStart` and `isAtEnd` were essentially testing before).

Reply:
Can we send a fake scroll event instead of overwriting internal state?